WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+        }
+
+        alertController.addActionWithTitle(NSLocalizedString("Cancel", comment: ""), style: .cancel)
+        present(alertController, animated: true, completion: nil)
+
+        // TODO: replace the alert implementation with something nicer like `LinkSettingsViewController` on WPiOS.
+
+        richTextView.resignFirstResponder()
+    }
+
+    func insertLink(url: String, text: String?, target: String?, range: NSRange) {
+        let linkURLString = url
+        var linkText = text
+
+        if linkText == nil || linkText!.isEmpty {
+            linkText = linkURLString
+        }
+
+        guard let url = URL(string: linkURLString), let title = linkText else {
+            return
+        }
+
+        richTextView.setLink(url.normalizedURLForWordPressLink(), title: title, target: target, inRange: range)
+    }
+
+    func removeLink(in range: NSRange) {
+//        trackFormatBarAnalytics(stat: .editorTappedUnlink)
+        richTextView.removeLink(inRange: range)
+    }
+
+    func toggleHeader(fromItem item: FormatBarItem) {
+        guard !optionsTablePresenter.isOnScreen() else {
+            optionsTablePresenter.dismiss()
+            return
+        }
+
+//        trackFormatBarAnalytics(stat: .editorTappedHeader)
+
+        let headerOptions = Constants.headers.map { headerType -> OptionsTableViewOption in
+            let attributes: [NSAttributedString.Key: Any] = [
+                .font: UIFont.systemFont(ofSize: CGFloat(headerType.fontSize)),
+            ]
+
+            let title = NSAttributedString(string: headerType.description, attributes: attributes)
+
+            return OptionsTableViewOption(image: headerType.iconImage,
+                                          title: title,
+                                          accessibilityLabel: headerType.accessibilityLabel)
+        }
+
+        let selectedIndex = Constants.headers.firstIndex(of: self.headerLevelForSelectedText())
+
+        let optionsTableViewController = OptionsTableViewController(options: headerOptions)
+        optionsTableViewController.applyDefaultStyles()
+
+        optionsTablePresenter.present(
+            optionsTableViewController,
+            fromBarItem: item,
+            selectedRowIndex: selectedIndex,
+            onSelect: { [weak self] selected in
+                guard let range = self?.richTextView.selectedRange else { return }
+
+//                let selectedStyle = Analytics.headerStyleValues[selected]
+//                self?.trackFormatBarAnalytics(stat: .editorTappedHeaderSelection, headingStyle: selectedStyle)
+
+                self?.richTextView.toggleHeader(Constants.headers[selected], range: range)
+                self?.optionsTablePresenter.dismiss()
+        })
+    }
+
+    func insertHorizontalRuler() {
+//        trackFormatBarAnalytics(stat: .editorTappedHorizontalRule)
+        richTextView.replaceWithHorizontalRuler(at: richTextView.selectedRange)
+    }
+
+    func insertMore() {
```

Comment:
Most of these methods also look like pure functions, where we perform actions on the richTextView. Maybe there is potential for extracting them to another class / struct